src/better_lane_follow/src/LineDetect.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Created by: Raad Khan
+ * Created On: April 23, 2017
+ * Description: Takes in an image feed and generates lane lines.
+ * Usage:
+ * Subscribes to:
+ * Publishes to:
+ */
+
+#include <LineDetect.h>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/objdetect/objdetect.hpp>
+
+LineDetect::LineDetect() : initalLineDetectThreshold(200),
+                           white(255),
+                           initialWindowWidth(10),
+                           numVerticalSlice(10) {}
+
+std::vector<int> LineDetect::getHistogram(cv::Mat& windowImage) {
+
+    std::vector<int> histogram(windowImage.cols, 0);
+
+    for (int j = 0; j < windowImage.rows; j++) {
+        for (int i = 0; i < windowImage.cols; i++) {
+            int pixelValue = (int)windowImage.at<uchar>(i, j);
+            if (pixelValue == white) {
+                histogram[i]++;
+            }
+        }
+    }
+
+    return histogram;
+}
+
+std::vector<Polynomial2D> getLines(cv::Mat& binaryImage) {
+
+    std::vector<int> baseHistogram = getHistogram(binaryImage);
```

Comment:
`LineDetect::getHistogram` is defined, but this function should be `LineDetect::getLines`, not just `getLines`